db/Pos/DB/Functions.hs

```diff
@@ -41,6 +43,23 @@ dbGetBi tag key = do
 dbPutBi :: (Bi v, MonadDB m) => DBTag -> ByteString -> v -> m ()
 dbPutBi tag k v = dbPut tag k (serialize' v)
 
+-- | Read serialized value (with version) associated with given key from pure DB.
+dbGetBiWithVersion
+    :: forall v m.
+       (Bi v, MonadDBRead m)
+    => DBTag -> ByteString -> m (Maybe (Word8, v))
+dbGetBiWithVersion tag key = do
+    bytes <- dbGet tag key
+    traverse (dbDecode . (ToDecodeValue key)) bytes
```

Comment:
This helper can't be used as is. It should check that tag is the same as the one we expect and throw an exception if it differs (e. g. create a new constructor of `DBError`, like `DBUnexpectedTag`).

Reply:
Do you mean tag here, or version?

Reply:
Right, I meant version, but called it tag (in a certain sense it can be called tag, I guess, but in this case it's indeed confusing, because we have another tag).